### PR TITLE
Use $ReadOnlyArray<string> for validation errors

### PR DESCRIPTION
**Pull request by dmnd**
As requested by @flex-tjon [here][1]. Don't know why we weren't already 
doing this.

Also I had to edit a lot of places, so maybe there's opportunity for a type
alias, but that can be solved another day.

  [1]: https://github.com/flexport/flexport/pull/44287#discussion_r292715411

### CODE REVIEW

File: src/Form.js [hunk 1]
```diff
@@ -184,7 +184,7 @@ function getRelativePath(path: Path, prefix: Path) {
 function validateTree<T>(
   prefix: Path,
   [value, tree]: FormState<T>,
-  validations: Map<EncodedPath, Map<number, (mixed) => Array<string>>>
```
Comment (zgotsch) [inline src/Form.js]: In general, you should be strict in your input types and lax in your output types, since functions are contravariant in their arguments and covariant in their returns.

Why did you need to change these?

Comment (dmnd) [reply src/Form.js]: I changed everything because it's bad to mutate data stored in React props or state.

Comment (dmnd) [reply src/Form.js]: Ok, I pushed a new version that changes `externalErrors` and then only what is minimally necessary. Please take another look 🙏
